### PR TITLE
GH1536: Install empty CT log list

### PR DESCRIPTION
**Pull request by richsalz**
This replaces #1536 after we got consensus to just install an empty file.
For master.
Ping @levitte to see if I got the MMS stuff right, @RJPercival and @vdukhovni to check that it's doing the right thing based on our discussion.

### CODE REVIEW

File: util/ct_log_list.cnf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# This file specifies the Certificate Transparency logs
+# that are to be trusted.
+
+# This file is empty in the OpenSSL source distribution.  One way
+# to generate a file (such as from Chrome's trust list) is to
+# fetch and run this python script:
+# https://github.com/RJPercival/certificate-transparency/blob/0a753fb4487bbf33de7b202133aac95cd0bf21b2/python/utilities/log_list/openssl_generator.py
```
Comment (RJPercival) [inline util/ct_log_list.cnf]: Could you change this link to `https://github.com/google/certificate-transparency/blob/master/python/utilities/log_list/print_log_list.py` and mention using the argument `--openssl_output "ct_log_list.cnf"` please?

File: util/ct_log_list.cnf [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+# on the log list found at https://sites.google.com/site/certificatetransparency/known-logs.
```
Comment (RJPercival) [inline util/ct_log_list.cnf]: Could you make this link http://www.certificate-transparency.org/known-logs please?

**Comment by RJPercival** [comment]
Thanks for putting this together. Other than changing a couple of links, LGTM.

File: util/ct_log_list.cnf (hunk 2 above)
Comment (richsalz) [inline util/ct_log_list.cnf]: Great, done

File: util/ct_log_list.cnf (hunk 1 above)
Comment (richsalz) [inline util/ct_log_list.cnf]: I updated the comments, please take a lok.

File: util/ct_log_list.cnf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# This file specifies the Certificate Transparency logs
+# that are to be trusted.
+
+# This file is empty in the OpenSSL source distribution.  One way
+# to generate a file (such as from Chrome's trust list) is the following:
+#       wget https://raw.githubusercontent.com/google/certificate-transparency/master/python/utilities/log_list/print_log_list.py
+#       python print_log_list.py --openssl_output ct_log_list.cnf https://sites.google.com/site/certificatetransparency/known-logs.
```
Comment (RJPercival) [inline util/ct_log_list.cnf]: This wouldn't work. You'd have to wget the other Python files in that directory at the minimum. Plus, there are a bunch of other flags required for print_log_list.py to actually work. I think it'd be best to just:

- link to www.certificate-transparency.org/known-logs
- link to https://github.com/google/certificate-transparency/blob/master/python/utilities/log_list/print_log_list.py
- mention that the `--openssl_output` flag is required
- leave it as an exercise for the reader to figure out exactly how they invoke the command

That provides all of the pieces of the puzzle, and then the reader just needs to read `print_log_list.py --help`. It should be the responsibility of the CT project to improve their documentation for how that tool is used, if it isn't clear enough.

Comment (richsalz) [inline util/ct_log_list.cnf]: Sure, done.

File: util/ct_log_list.cnf [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# This file specifies the Certificate Transparency logs
+# that are to be trusted.
+
+# Google's list of logs can be found here:
+#       www.certificate-transparency.org/known-logs 
+# A Python program to parse the log can be found here:
```
Comment (RJPercival) [inline util/ct_log_list.cnf]: s/parse the log/convert the log list to OpenSSL's format/

**Comment by RJPercival** [comment]
One nit, otherwise looks great, thanks.

File: util/ct_log_list.cnf (hunk 4 above)
Comment (richsalz) [inline util/ct_log_list.cnf]: Good catch.

File: Configurations/unix-Makefile.tmpl [hunk 5]
```diff
@@ -355,6 +355,15 @@ install_ssldirs:
 		cp $(SRCDIR)/apps/openssl.cnf $(DESTDIR)$(OPENSSLDIR)/openssl.cnf; \
 		chmod 644 $(DESTDIR)$(OPENSSLDIR)/openssl.cnf; \
 	fi
+	@echo "intsall $(SRCDIR)/util/ct_log_list.cnf -> $(DESTDIR)$(OPENSSLDIR)/ct_log_list.cnf.dist"
+	@cp $(SRCDIR)/util/ct_log_list.cnf $(DESTDIR)$(OPENSSLDIR)/ct_log_list.cnf.new
+	@chmod 644 $(DESTDIR)$(OPENSSLDIR)/ct_log_list.cnf.new
+	@mv -f  $(DESTDIR)$(OPENSSLDIR)/ct_log_list.cnf.new $(DESTDIR)$(OPENSSLDIR)/ct_log_list.cnf.dist
+	@if ! [ -f "$(DESTDIR)$(OPENSSLDIR)/ct_log_list.cnf" ]; then \
```
Comment (dot-asm) [inline Configurations/unix-Makefile.tmpl]: Solaris shell will have problem with `if ! [ -f ... ]`. It's either `if [ ! -f ... ]` or `if [ -f ... ] then :; else`...

Comment (richsalz) [inline Configurations/unix-Makefile.tmpl]: Oh.  It was already used in openssl.cnf section just above.  Updated commit pushed; it fixes both.

**Comment by richsalz** [comment]
And this is for master and 1.1.0

File: Configurations/unix-Makefile.tmpl [hunk 6]
```diff
@@ -350,11 +350,20 @@ install_ssldirs:
 	@cp $(SRCDIR)/apps/openssl.cnf $(DESTDIR)$(OPENSSLDIR)/openssl.cnf.new
 	@chmod 644 $(DESTDIR)$(OPENSSLDIR)/openssl.cnf.new
 	@mv -f  $(DESTDIR)$(OPENSSLDIR)/openssl.cnf.new $(DESTDIR)$(OPENSSLDIR)/openssl.cnf.dist
-	@if ! [ -f "$(DESTDIR)$(OPENSSLDIR)/openssl.cnf" ]; then \
+	@if [ ! -f "$(DESTDIR)$(OPENSSLDIR)/openssl.cnf" ]; then \
 		echo "install $(SRCDIR)/apps/openssl.cnf -> $(DESTDIR)$(OPENSSLDIR)/openssl.cnf"; \
 		cp $(SRCDIR)/apps/openssl.cnf $(DESTDIR)$(OPENSSLDIR)/openssl.cnf; \
 		chmod 644 $(DESTDIR)$(OPENSSLDIR)/openssl.cnf; \
 	fi
+	@echo "intsall $(SRCDIR)/util/ct_log_list.cnf -> $(DESTDIR)$(OPENSSLDIR)/ct_log_list.cnf.dist"
```
Comment (levitte) [inline Configurations/unix-Makefile.tmpl]: install

**Comment by levitte** [comment]
I will check this out on VMS tonight or tomorrow morning. 

One thing, though... Why in `util`?

File: Configurations/unix-Makefile.tmpl (hunk 6 above)
Comment (richsalz) [inline Configurations/unix-Makefile.tmpl]: Fxied ☺

**Comment by richsalz** [comment]
Could not think of a better place.  Apps?

**Comment by levitte** [comment]
Considering that's where we have `openssl.cnf`, then yes, that's where I'd expect to find the CT log file as well. 

Also, `util` is our internal utilities. Putting a file to be installed there is just asking for confusion...

**Comment by levitte** [comment]
One last comment...  the `.dist` copy seems like a nice feature, but should probably be a separate PR and affect `openssl.cnf` as well, wouldn't you think?

**Comment by richsalz** [comment]
The .dist copy was in the makefile; I didn't add it.  I just used it :)

**Comment by richsalz** [comment]
updated commit to put the conf file in apps

File: Configurations/descrip.mms.tmpl [hunk 7]
```diff
@@ -383,6 +383,12 @@ install_ssldirs : check_INSTALLTOP
         IF F$SEARCH("OSSL_DATAROOT:[000000]openssl.cnf") .EQS. "" THEN -
                 COPY/PROT=W:R {- sourcefile("apps", "openssl-vms.cnf") -} -
                         ossl_dataroot:[000000]openssl.cnf
+        @ ! Install CTLOG configuration file
+        COPY/PROT=W:R {- sourcefile("util", "ct_log_list.cnf") -} -
```
Comment (levitte) [inline Configurations/descrip.mms.tmpl]: apps

File: apps/ct_log_list.cnf [hunk 8]
```diff
@@ -1,34 +1,9 @@
...
-key = MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEJ2Q5DC3cUBj4IQCiDu0s6j51up+TZAkAEcQRF6tczw90rLWXkJMAW7jr9yc92bIKgV8vDXU4lDeZHvYHduDuvg==
-
-[symantec]
-description = Symantec log
-key = MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEluqsHEYMG1XcDfy1lCdGV0JwOmkY4r87xNuroPS2bMBTP01CEDPwWJePa75y9CrsHEKqAy8afig1dpkIPSEUhg==
-
-[venafi]
-description = Venafi log
-key = MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAolpIHxdSlTXLo1s6H1OCdpSj/4DyHDc8wLG9wVmLqy1lk9fz4ATVmm+/1iN2Nk8jmctUKK2MFUtlWXZBSpym97M7frGlSaQXUWyA3CqQUEuIJOmlEjKTBEiQAvpfDjCHjlV2Be4qTM6jamkJbiWtgnYPhJL6ONaGTiSPm7Byy57iaz/hbckldSOIoRhYBiMzeNoA0DiRZ9KmfSeXZ1rB8y8X5urSW+iBzf2SaOfzBvDpcoTuAaWx2DPazoOl28fP1hZ+kHUYvxbcMjttjauCFx+JII0dmuZNIwjfeG/GBb9frpSX219k1O4Wi6OEbHEr8at/XQ0y7gTikOxBn/s5wQIDAQAB
-
```
Comment (levitte) [inline apps/ct_log_list.cnf]: Is this removal going to disturb some test?

**Comment by levitte** [comment]
> The .dist copy was in the makefile; I didn't add it.  I just used it :)

You're right.  I was blind.  Sorry...

File: Configurations/descrip.mms.tmpl (hunk 7 above)
Comment (richsalz) [inline Configurations/descrip.mms.tmpl]: Good catch.  Updated commit pushed.

File: apps/ct_log_list.cnf (hunk 8 above)
Comment (richsalz) [inline apps/ct_log_list.cnf]: No, all tests pass.  The CT tests have their own conf file in the test directory.

**Comment by levitte** [comment]
VMS runs ok.  Have you noticed, though, that Travis complains in two specific builds?  That's worth investigating.

**Comment by richsalz** [comment]
Travis fails on master.  Will look at it.  Okay to ship this?

**Comment by levitte** [comment]
Yup, ship

**Pull request closed without merging**